openapi_server/apis/reviews_api.py

```diff
@@ -42,7 +54,88 @@ async def add_review(
+    try:
+        restaurant: DbRestaurant = DbRestaurant.get(create_review.restaurant_id)
+    except DbRestaurant.DoesNotExist:
+        try:
+            async with AsyncClient(base_url="https://maps.googleapis.com") as ac:
+                response = await ac.get(
+                    f"/maps/api/place/details/json?place_id={create_review.restaurant_id}&key={os.environ.get('GOOGLE_API_KEY')}"
+                )
+                result: Dict = response.json()["result"]
+
+            restaurant = DbRestaurant(result["place_id"])
+            restaurant.update(
+                actions=[
+                    DbRestaurant.name.set(result["name"]),
+                    DbRestaurant.latitude.set(result["geometry"]["location"]["lat"]),
+                    DbRestaurant.longitude.set(result["geometry"]["location"]["lng"]),
+                    DbRestaurant.address.set(result["formatted_address"]),
+                ]
+            )
+            restaurant.save()
+        except Exception as e:
+            raise HTTPException(status_code=500, detail=str(e))
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    new_review: DbReview = DbReview(uuid4().hex)
+    new_review.update(
+        actions=[
+            DbReview.created_at.set(str(datetime.utcnow())),
+            DbReview.username.set(create_review.username),
+            DbReview.restaurantId.set(create_review.restaurant_id),
+            DbReview.rating.set(create_review.rating),
+            DbReview.favorite_food.set(create_review.favorite_food),
+            DbReview.starred.set(create_review.starred),
+        ]
+    )
+    actions: List[Action] = [DbReview.updated_at.set(new_review.created_at)]
+    # optional request body properties
+    if create_review.content:
+        actions.append(DbReview.content.set(create_review.content))
+    if create_review.photo_url:
+        actions.append(DbReview.photo_url.set(create_review.photo_url))
+
+    new_review.update(actions)
+    try:
+        new_review.save()
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+    return Review(
+        id=new_review.id,
+        user=User(
+            id=user.id,
+            username=user.username,
+            firstName=user.first_name,
+            lastName=user.last_name,
+            email=user.email,
+            password=user.password,
+        ),
+        restaurant=Restaurant(
+            id=restaurant.id,
+            name=restaurant.name,
+            latitude=restaurant.latitude,
+            longitude=restaurant.longitude,
+            address=restaurant.address,
+        ),
+        rating=new_review.rating,
+        content=new_review.content,
+        photoUrl=new_review.photo_url,
+        favoriteFood=new_review.favorite_food,
+        starred=new_review.starred,
+        createdAt=new_review.created_at,
+        updatedAt=new_review.updated_at,
+    )
```

Comment:
instead of dealing with the case conversion, we could probably use such https://pypi.org/project/fastapi-camelcase/ library to convert the cases and reduce the efforts to deal with the case conversion.